libkbfs/md_journal.go

```diff
@@ -1183,3 +1183,104 @@ func (j *mdJournal) clear(
 	}
 	return nil
 }
+
+func (j *mdJournal) resolveAndClear(
+	ctx context.Context, signer cryptoSigner, ekg encryptionKeyGetter,
+	bsplit BlockSplitter, bid BranchID, rmd *RootMetadata) (
+	mdID MdID, err error) {
+	j.log.CDebugf(ctx, "Resolve and clear, branch %s, resolve rev %d",
+		bid, rmd.Revision())
+	defer func() {
+		if err != nil {
+			j.deferLog.CDebugf(ctx,
+				"Resolving journal for branch %s failed with %v",
+				bid, err)
+		}
+	}()
+
+	// The resolution must not have a branch ID.
+	if rmd.BID() != NullBranchID {
+		return MdID{}, fmt.Errorf("Resolution MD has branch ID: %s", rmd.BID())
+	}
+
+	// The branch ID must match our current state.
+	if bid == NullBranchID {
+		return MdID{}, errors.New("Cannot resolve master branch")
+	}
+	if j.branchID != bid {
+		return MdID{}, fmt.Errorf("Resolve and clear for branch %s "+
+			"while on branch %s", bid, j.branchID)
+	}
+
+	// First make a new journal to hold the block.
+	otherJournal, err := makeMDJournal(j.uid, j.key, j.codec, j.crypto,
```

Comment:
Can you make a helper function `makeMDJournalWithIDJournal` that takes an `mdIDJournal` and that `makeMDJournal` delegates to? Then you can make the temp one and pass it in. As a bonus, the branch ID will then automatically be null.